src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+
+M['wal.ext.spaces.*.old'] = M['wal.ext.old']
+
+M['wal.max_size'] = format_text([[
+    The maximum number of bytes in a single write-ahead log file. When a
+    request would cause an `.xlog` file to become larger than `wal.max_size`,
+    Tarantool creates a new WAL file.
+]])
+
+M['wal.mode'] = format_text([[
+    Specify fiber-WAL-disk synchronization mode as: none: write-ahead log is
+    not maintained. A node with wal.mode set to none can't be a replication
+    master. write: fibers wait for their data to be written to the
+    write-ahead log (no fsync(2)). fsync: fibers wait for their data,
+    fsync(2) follows each write(2).
+
+]])
+
+M['wal.queue_max_size'] = format_text([[
+    The size of the queue in bytes used by a replica to submit new transactions
+    to a write-ahead log (WAL). This option helps limit the rate at which a
+    replica submits transactions to the WAL. Limiting the queue size might be
+    useful when a replica is trying to sync with a master and reads new
+    transactions faster than writing them to the WAL.
+]])
+
+M['wal.retention_period'] = format_text([[
+    The delay in seconds used to prevent the Tarantool garbage collector from
+    removing a write-ahead log file after it has been closed. If a node is
+    restarted, `wal.retention_period` counts down from the last modification
+    time of the write-ahead log file.
+
+    The garbage collector doesn't track write-ahead logs that are to be
+    relayed to anonymous replicas, such as:
+
+    - Anonymous replicas added as a part of a cluster configuration
+    (see `replication.anon`).
+    -CDC (Change Data Capture) that retrieves data using anonymous replication.
+
+    In case of a replica or CDC downtime, the required write-ahead logs
+    can be removed. As a result, such a replica needs to be rebootstrapped.
+    You can use wal.retention_period to prevent such issues.
+
+    Note that `wal.cleanup_delay` option also sets the delay used to prevent
+    the Tarantool garbage collector from removing write-ahead logs. The
+    difference is that the garbage collector doesn't take into account
+    `wal.cleanup_delay` if all the nodes in the replica set are up and running,
+    which may lead to the removal of the required write-ahead logs.
+]])
+
+-- }}} wal configuration
+
+-- }}} Instance descriptions
+
+-- {{{ Cluster descriptions
+
+local L = {}
+
+L[''] = 'Cluster configuration'
+
+-- {{{ conditional configuration
+
+L['conditional'] = format_text([[
+    The `conditional` section defines the configuration parts that apply to
+    instances that meet certain conditions.
+]])
+
+L['conditional.*'] = format_text([[
+    `conditional.if`: Defines a section where configuration options apply only
+    if specified conditions are met.
+
+    Conditions can include `tarantool_version` and support comparison operators
+    (`>`, `<`, `>=`, `<=`, `==`, `!=`), as well as logical operators
+    (`||`, `&&`) and parentheses for grouping.
+]])
```

Comment:
This description defines the `if` field, but the element can contains all the fields from the cluster configuration (except `conditional`). I think that describing just `if` would be misleading.

I would write something like the following:

> Contains a part of the cluster configuration that is applied only if the specified conditions are met. Usually used to apply options that exist only on particular tarantool versions.
>
> The fields in this mapping are the same as in the cluster configuration, except:
>
> * The special `if` field holds the condition.
>
>   Conditions can include `tarantool_version`, three-digit version literals (`3.2.1`) and support comparison operators (`>`, `<`, `>=`, `<=`, `==`, `!=`), as well as logical operators (`||`, `&&`) and parentheses for grouping.
>
> * Has no `conditional` section.

Note: added the usage scenario: just 'some conditions' is too vague for my taste.

Note: added information about the version literals. It is described in details in #9461.

Reply:
Replaced with
```lua
C['conditional.*'] = format_text([[
    Contains a part of the cluster configuration that is applied only if
    the specified conditions are met. Usually used to apply options that
    exist only on particular tarantool versions.

    The fields in this mapping are the same as in the cluster configuration,
    except:

    - The special `if` field holds the condition.
    - Conditions can include `tarantool_version`, three-digit version
      literals (`3.2.1`) and support comparison operators (`>`, `<`, `>=`,
      `<=`, `==`, `!=`), as well as logical operators (`||`, `&&`) and
      parentheses for grouping.
    - Has no `conditional` section.
]])
```

Reply:
OK for me except that I would replace

```markdown
- The special `if` field holds the condition.
- Conditions <...>
```

with

```markdown
- The special `if` field holds the condition.

  Conditions <...>
```

Or move the `if` conditions descriptions out of the list of the differences from usual cluster ocnfiguration fields.

Otherwise it gives a false impression that these 'conditions' are something completely different from the `if` clause.

Reply:
Now it should be as intended.